File: apps/web-api/src/EasyLab.Core/Entities/Course.cs
using System;
using System.Collections.Generic;

namespace EasyLab.Core.Entities
{
    public class Course
    {
        public Course(){
            Assignments = new List<Assignment>();
            Users = new List<CourseUser>();
            DateCreated = DateTime.UtcNow;
        }
        public Guid Id { get; set; }

        public string Name { get; set; }

        public string Description { get; set; }

        public bool IsActive { get; set; }

        public DateTime DateCreated { get; set; }

        public Guid AuthorId { get; set; }


        public virtual User Author { get; set; }

        public virtual List<Assignment> Assignments { get; set; }

        public virtual List<CourseUser> Users { get; set; }
    }
}

File: apps/web-ui/src/app/data/schema/course.ts
export class Course {

    constructor(id:string,name:string,description:string,dateCreated:Date,studentsEnrolled:number){
        this.id = id;
        this.name = name;
        this.description = description;
        this.dateCreated = dateCreated;
        this.studentsEnrolled = studentsEnrolled;
    }

    public id: string;
    public name: string;
    public description: string;
    public dateCreated: Date;
    public studentsEnrolled: number;
}
